control-plane/subcommand/server-acl-init/command.go

```diff
@@ -804,6 +765,91 @@ func (c *Command) createAuthMethod(consulClient *api.Client, authMethod *api.ACL
 		})
 }
 
+type gatewayRulesGenerator func(name, namespace string) (string, error)
+
+// ConfigureGatewayParams are parameters used to configure Ingress and Terminating Gateways.
+type ConfigureGatewayParams struct {
+	//GatewayType specifies whether it is an ingress or terminating gateway
+	GatewayType string
+	//GatewayNames is the collection of gateways that have been specified.
+	GatewayNames []string
+	//AuthMethodName is the authmethod for which to register the binding rules and policies for the gateways
+	AuthMethodName string
+	//RuleGenerator is the function that supplies the rules that will be added to the policy.
+	RulesGenerator gatewayRulesGenerator
+	//ConsulDC is the name of the DC where the gateways will be registered
+	ConsulDC string
+	//PrimaryDC is the name of the Primary Data Center
+	PrimaryDC string
+	//Primary specifies whether the ConsulDC is the Primary Data Center
+	Primary bool
+}
+
+func (c *Command) configureGateway(gatewayParams ConfigureGatewayParams, consulClient *api.Client) error {
+	// Create a token for each gateway name. Each gateway needs a
+	// separate token because users may need to attach different policies
+	// to each gateway token depending on what the services it represents
```

Comment:
```suggestion
	// Each gateway needs to be configured
	// separately because users may need to attach different policies
	// to each gateway role depending on what services it represents.
```